frontend/public/components/source-to-image.tsx

```diff
@@ -0,0 +1,507 @@
+      kind: 'Route',
+      apiVersion: 'route.openshift.io/v1',
+      metadata: {
+        name,
+        namespace,
+        labels,
+      },
+      spec: {
+        to: {
+          kind: 'Service',
+          name,
+        },
+        // The service created by `createService` uses the same port as the container port.
+        port: {
+          // Use the port name, not the number for targetPort. The router looks
+          // at endpoints, not services, when resolving ports, so port numbers
+          // will not resolve correctly if the service port and container port
+          // numbers don't match.
+          targetPort: makePortName(firstPort),
+        },
+        wildcardPolicy: 'None',
+      },
+    };
+
+    return k8sCreate(RouteModel, route);
+  }
+
+  handleError = err => {
+    this.setState({error: this.state.error ? `${this.state.error}; ${err.message}` : err.message});
+  }
+
+  save = (event: React.FormEvent<EventTarget>) => {
+    event.preventDefault();
+    const { namespace, selectedTag, name, repository, createRoute, ports } = this.state;
+    if (!name || !selectedTag || !namespace || !repository) {
+      this.setState({error: 'Please complete all fields.'});
+      return;
+    }
+
+    const requests = [
+      this.createDeploymentConfig(),
+      this.createImageStream(),
+      this.createBuildConfig(),
+    ];
+
+    // Only create a service or route if the builder image has ports.
+    if (!_.isEmpty(ports)) {
+      requests.push(this.createService());
+      if (createRoute) {
+        requests.push(this.createRoute());
+      }
+    }
+
+    requests.forEach(r => r.catch(this.handleError));
+    this.setState({ inProgress: true, error: null });
+    Promise.all(requests).then(() => {
+      this.setState({inProgress: false});
+      if (!this.state.error) {
+        history.push(resourcePathFromModel(DeploymentConfigModel, this.state.name, this.state.namespace));
+      }
+    }).catch(() => this.setState({inProgress: false}));
+  };
+
+  render() {
+    const { obj } = this.props;
+    const { selectedTag, ports } = this.state;
+    if (obj.loadError) {
+      return <LoadError message={obj.loadError.message} label="Image Stream" className="loading-box loading-box__errored" />;
+    }
+
+    if (!obj.loaded) {
+      return <LoadingBox />;
+    }
+
+    const imageStream = obj.data;
+    const { tags } = this.state;
+    if (_.isEmpty(tags)) {
+      return <MsgBox title="No Builder Tags" detail={`ImageStream ${imageStream.metadata.name} has no Source-to-Image builder tags.`} />;
+    }
+
+    const tag = _.find(imageStream.spec.tags, { name: selectedTag });
+    const sampleRepo = getSampleRepo(tag);
+
+    // TODO: Sort tags by semver.
+    const tagOptions = {};
+    _.each(imageStream.spec.tags, ({name}) => tagOptions[name] = <ResourceName kind="ImageStreamTag" name={`${imageStream.metadata.name}:${name}`} />);
+
+    return <React.Fragment>
+      <div className="co-m-pane__body">
```

Comment:
Sorry.  I should have caught this earlier.  This `.co-m-pane__body` is not needed and is resulting in extra padding.
![screen shot 2018-09-18 at 3 38 46 pm](https://user-images.githubusercontent.com/895728/45711950-11357680-bb59-11e8-8b24-2fa6c32e35d1.PNG)